tools/autograd/templates/Functions.cpp

```diff
@@ -74,6 +74,29 @@ Tensor maybe_multiply(const Tensor & t, const Scalar & s) {
   }
 }
 
+std::tuple<Tensor, Tensor> index_add_backward(const Tensor & grad, int64_t dim, const Tensor & index, const Tensor & weight) {
+
+  // grad for source
+  Tensor grad_source = grad.index_select(dim, index);
+
+  bool defined = weight.defined();
+  Tensor weighted;
+  Tensor ind;
+  if(!defined){
+    return std::tie(grad,grad_source);
+  }else{
+    /* TO-DO: tile gracefully  */
```

Comment:
Is defined a good name here?
Shouldn't this be just something like grad_source = grad_source * weight.view(-1, ones_to_match_weight_dim) or so? I'm not sure what to think of the loop.

Reply:
Right, that loop is one ugly mess.. view shall do the trick. Thank You.